### ISSUE TITLE
pinging: driver: bad connection

### ISSUE THREAD

**Issue by achettyiitr** [databricks/databricks-sql-go#161]
```GO
func TestDatabricks(t *testing.T) {
	const (
		host      = "xxx"
		port      = 443
		path      = "xxx"
		token     = "xxx"
		userAgent = "xxx"
		catalog   = ""
	)

	connector, err := dbsql.NewConnector(
		dbsql.WithServerHostname(host),
		dbsql.WithPort(port),
		dbsql.WithHTTPPath(path),
		dbsql.WithAccessToken(token),
		dbsql.WithSessionParams(map[string]string{
			"ansi_mode": "false",
		}),
		dbsql.WithUserAgentEntry(userAgent),
		dbsql.WithInitialNamespace(
			catalog,
			"",
		),
		dbsql.WithTimeout(time.Second*15),
		dbsql.WithRetries(10, 1*time.Second, 300*time.Second),
	)
	require.NoError(t, err)

	db := sql.OpenDB(connector)
	defer func() { _ = db.Close() }()

	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
	defer cancel()

	err = db.PingContext(ctx)
	require.NoError(t, err)
}
```

Issue: Getting driver bad connection, while calling Ping with `ContextWithTimeout`. Although, it's not reproducible. There is some edge case happening when the cluster is starting and we are executing Query.
Driver Version: v1.1.0
Expectation: To return the error instead of DriverBadConnection.

![Screenshot 2023-09-19 at 10 51 44 PM (2)](https://github.com/databricks/databricks-sql-go/assets/77991816/899f8538-a7d8-45a6-85ca-0bf647e3c2e7)
![Screenshot 2023-09-19 at 10 50 29 PM](https://github.com/databricks/databricks-sql-go/assets/77991816/66e3faab-c080-4944-bda4-e5a45ba95113)

**Comment by rcypher-databricks** [comment]
Thanks for raising this issue. We will look into a change.

**Comment by rcypher-databricks** [comment]
In version 1.5.1 we are now wrapping the causative error in a bad connection error.  You can use errors.Is/As to check for timeout, cancellation, etc.

**Comment by rcypher-databricks** [comment]
Has the fix in 1.5.1 resolved this issue for you?
